### PR TITLE
Added extended type checking with the libtorrent API

### LINKED ISSUE #40: Generate libtorrent typing stubs
With the introduction of type checking in Tribler Experimental, it is becoming painfully obvious that Python does not understand libtorrent's Boost types.

If we continue like this, long-term we will need to (project-wide!) either ignore typing 😞, cast everything 😞, or provide type stubs manually 😞. This would make me a very sad developer.

It would be much more convenient if we had a tool to analyze and convert libtorrent's Boost types (this information is already available). The result should be Python type stub files (`.pyi`).

I'll start with a proof-of-concept and report back.

### PR DESCRIPTION
**Pull request by qstokkink**
Fixes #40

This PR:

 - Adds fetching the libtorrent type stubs when performing the unittest mypy check.
 - Fixes the typing errors related to libtorrent.

Note that this PR also makes TriblerExperimental libtorrent 2.X compatible. At least, according to the typing.

### CODE REVIEW

**Comment by qstokkink** [comment]
validate

File: src/tribler/core/libtorrent/download_manager/download.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, Awaitable, Callable, List, Tuple, TypedDict, cast
 
+import libtorrent
```
Comment (egbertbouman) [inline src/tribler/core/libtorrent/download_manager/download.py]: You're importing libtorrent twice.

File: src/tribler/core/libtorrent/download_manager/download_manager.py [hunk 2]
```diff
@@ -17,8 +17,9 @@
 from copy import deepcopy
 from pathlib import Path
 from tempfile import TemporaryDirectory
-from typing import TYPE_CHECKING, Any, Awaitable, Callable, Dict, Iterable, List
+from typing import TYPE_CHECKING, Any, Awaitable, Callable, Dict, Iterable, List, cast
 
+import libtorrent
```
Comment (egbertbouman) [inline src/tribler/core/libtorrent/download_manager/download_manager.py]: Duplicate import

**Comment by qstokkink** [comment]
@egbertbouman thanks! Fixed.

**Review by egbertbouman — APPROVED** [review]

**Pull request merged**